src/test/java/co/rsk/bitcoinj/script/P2shErpFederationRedeemScriptParserTest.java

```diff
@@ -327,6 +344,56 @@ public void isP2shErpFed_falseWithCustomRedeemScript() {
         Assert.assertFalse(P2shErpFederationRedeemScriptParser.isP2shErpFed(customRedeemScript.getChunks()));
     }
 
+    @Test
+    public void getMultiSigType_shouldReturnP2shErpFedType() {
+        Assert.assertEquals(MultiSigType.P2SH_ERP_FED, p2shErpFederationRedeemScriptParser.getMultiSigType());
+    }
+
+    @Test
+    public void getM_shouldReturnM() {
+        int expectedM = defaultRedeemScriptKeys.size() / 2 + 1;
+        Assert.assertEquals(expectedM, p2shErpFederationRedeemScriptParser.getM());
+    }
+
+    @Test
+    public void findKeyInRedeem_whenKeyExists_shouldReturnIndex() {
+        for (int i = 0; i < defaultRedeemScriptKeys.size(); i++) {
+            BtcECKey expectedKey = defaultRedeemScriptKeys.get(i);
+            Assert.assertEquals(i, p2shErpFederationRedeemScriptParser.findKeyInRedeem(expectedKey));
+        }
+    }
+
+    @Test
+    public void findKeyInRedeem_whenKeyDoesNotExists_shouldThrowIllegalStateException() {
+        BtcECKey unknownKey = BtcECKey.fromPrivate(BigInteger.valueOf(1234567890L));
+        for (int i = 0; i < defaultRedeemScriptKeys.size(); i++) {
+            try {
+                p2shErpFederationRedeemScriptParser.findKeyInRedeem(unknownKey);
+            } catch (Exception ex) {
+                Assert.assertTrue(ex instanceof IllegalStateException);
+                Assert.assertTrue(ex.getMessage().contains("Could not find matching key " + unknownKey
+                    + " in script "));
+            }
+        }
+    }
+
+    @Test
+    public void getPubKeys_shouldReturnPubKeys() {
+        List<BtcECKey> actualPubKeys = p2shErpFederationRedeemScriptParser.getPubKeys();
+        List<BtcECKey> expectedPubKeys = defaultRedeemScriptKeys.stream()
```

Comment:
Is this necessary? Why not use `defaultRedeemScriptKeys` directly?

Reply:
Yes, it is necessary. The pubic keys from `pubKeys` do not include the private, while the `defaultRedeemScriptKeys` list does. So the assertion failed when using the defaultRedeemScriptKeys because find that the keys without the private keys are different.